scalableminds/webknossos
# Define CSP

This PR introduces a CSP. The main changes are:
- All parameters used in the inline scripts in the main.html were put into meta tags
- Inline scripts were moved into js/ts files where possible (loading the initial style sheet).
- The inline scripts read the parameters from the meta tags. Thus the code inside the meta tags should be static and therefore the hash of the code should also be static. But this needs testing! The hashes of these inline scripts are included in the CSP to allow them.


### URL of deployed dev instance (used for testing):
- https://definecsp.webknossos.xyz

### Steps to test:
- This PR defines a reasonably narrow CSP for wk :).
- Check whether the CSP works on a dev branch.
- Check whether the CSP works with the connectome viewer. I tried this in my local dev env and it seemed to work.

### TODOs:
- [x] Internal Diskussion
- [x] Check connectome viewer
  - I checked this locally and everything seemed to work. But after merging we should recheck whether the connectome viewer still works.

### Issues:
- fixes #7345
- fixes #3085

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated migrations log
- [ ] After merging and deploying check whether the connectome viewer still works
